### PR TITLE
[feat] Home Page & General React App setup

### LINKED ISSUE #3: Create Routing and general React App
Expected features to be implemented:

1.  Create routing for all pages by implementing React router
2.  Setup all pages of Ecommerce, and add basic structures
3.  Setup styles with component library

### PR DESCRIPTION
**Pull request by Hamdrive**
## Overview 
This PR refers to issue #3 

As per the requirement of the project, it has now been converted to a fully fledged react application.

Features/Changes implemented in this PR:

1. Setup basic routing for the ecommerce app using React Router
2. Converted the Home Page into full fledged React components
3. Implementation of CSS modules to handle specificity with styles
4. Implementation of mock backend to fetch the categories data dynamically

As of now the React app is properly functioning as a desktop app, but also has some features of mobile app.
Lateron it will also be hosted on Netlify

## Preview
image

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,43 +1,15 @@
-import "./App.css";
-import logo from "./logo.png";
+import {Routes, Route} from "react-router-dom"
+import {Home, Products, Cart, Wishlist, Login, Signup} from "./pages/index.jsx"
```
Comment (shivampandey0) [inline src/App.js]: `./pages/index.jsx` is unnecessary, only `./pages` will do the same job

**Review by shivampandey0 — APPROVED** [review]
Code looks good with react modular practices.
Added a Suggestion.

File: src/App.js (hunk 1 above)
Comment (Hamdrive) [reply src/App.js]: Yes by React's inbuilt nature it will look for the index file, have update the imports

File: src/pages/home/HeroSection.jsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { Link } from "react-router-dom";
+import styles from "./Home.module.css";
+import "../../styles/utils.css";
+
+export default function HeroSection() {
+  return (
+    <>
```
Comment (lannisterrr) [inline src/pages/home/HeroSection.jsx:7]: No need to use react fragments , when you are returning a single div.

**Review by lannisterrr — COMMENTED** [review]
Good work, learned about how to export using a single file. Made on suggestion.

File: src/pages/home/HeroSection.jsx (hunk 2 above)
Comment (Hamdrive) [reply src/pages/home/HeroSection.jsx:7]: Yes that's correct but I am returning a single parent component hence why I'm using react fragments

**Comment by Hamdrive** [comment]
PR successfully reviewed by 2 reviewers, and changes suggested have been made.
Merging to dev branch and closing branch

**Pull request merged**